scripts/install/configuration/apply_csm_configuration.sh

```diff
@@ -0,0 +1,174 @@
+{
+   # Display Help
+   echo "Runs CFS to setup passwordless ssh"
+   echo "All parameters are optional and the values will be determined automatically if not set."
+   echo
+   echo "Usage: deploy_ssh_keys.sh [ --csm-release version ] [ --git-commit id ]"
+   echo "                            [ --git-clone-url url ] [ --ncn-config-file file ]"
+   echo "                            [ --xnames xname1,xname2... ] [ --clear-state ]"
+   echo
+   echo "Options:"
+   echo "csm-release          The version of the CSM release to use."
+   echo "git-commit           The git commit id for CFS to use."
+   echo "git-clone-url        The git clone url for CFS to use."
+   echo "ncn-config-file      A file containing the NCN CFS configuration."
+   echo "xnames               A comma seperated list xnames to deploy to. All management nodes will be included if not set."
+   echo "clear-state          Clears existing state from components to ensure CFS runs."
+   echo
+}
+
+if [[ $# -eq 0 ]]; then
+  usage
+fi
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    --csm-release)
+      RELEASE="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --git-commit)
+      COMMIT="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --git-clone-url)
+      CLONE_URL="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --ncn-config-file)
+      NCN_CONFIG_FILE="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --xnames)
+      XNAMES="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --clear-state)
+      CLEAR_STATE="true"
+      shift # past argument
+      ;;
+    -h|--help) # help option
+      usage
+      exit 0
+      ;;
+    *) # unknown option
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+## CONFIGURATION SETUP ##
+if [[ -z "${RELEASE}" ]]; then
+    RELEASE=$(kubectl -n services get cm cray-product-catalog -o jsonpath='{.data.csm}' 2>/dev/null\
+        | yq r -j - 2>/dev/null | jq -r ' to_entries | .[0].value.configuration.import_branch' 2>/dev/null\
+        | awk -F'/' '{ print $NF }')
+    echo "Found release version ${RELEASE}"
+fi
+
+CSM_CONFIG_FILE="csm-config-$RELEASE.json"
+if [[ -z "${NCN_CONFIG_FILE}" ]]; then
+    NCN_CONFIG_FILE="ncn-personalization.json"
+fi
+if [[ -z "${CLONE_URL}" ]]; then
+    CLONE_URL="https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
+fi
+
+if [[ -z "${COMMIT}" ]]; then
+    VCS_USER=$(kubectl get secret -n services vcs-user-credentials --template={{.data.vcs_username}} | base64 --decode)
+    VCS_PASSWORD=$(kubectl get secret -n services vcs-user-credentials --template={{.data.vcs_password}} | base64 --decode)
+    echo "https://${VCS_USER}:${VCS_PASSWORD}@api-gw-service-nmn.local/vcs/cray/csm-config-management.git" > .git-credentials
+    echo "${CLONE_URL/\/\//\/\/${VCS_USER}:${VCS_PASSWORD}@}" > .git-credentials
+    git config --file .gitconfig credential.helper store
```

Comment:
Any worries here about stomping on a customer's previous settings?

Reply:
This is supposed to be an install script, so I wouldn't expect an settings to stomp on.